### ISSUE TITLE
ComponentPlayground – Maximum update depth exceeded

### ISSUE THREAD

**Issue by lennerd** [FormidableLabs/spectacle#617]
### Describe Your Environment

What version of Spectacle are you using? (can be found by running `npm list spectacle`)
spectacle@5.2.2

What version of React are you using? (can be found by running `npm list react`)
react@16.6.3

What browser are you using?
Chrome@70.0.3538.110

What machine are you on?
macOS@10.14.1

### Describe the Problem

When using the `ComponentPlayground` component with custom code an endless update cycle is started.

> Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.

Like the error message describes the problem, the `ComponentPlayground` is updating itself due enhancing the scope with `Component` class from React. When `getDerivedStateFrom` is comparing both previous scope (inside state) and the new one (from props), it actually compares my custom scope with the enhanced one. This of course will never be true and leads to the endless loop described above.

``` js
  static getDerivedStateFromProps(nextProps, prevState) {
    const updatedState = {};
    if (nextProps.code !== prevState.code) {
      const code = (nextProps.code || defaultCode).trim();
      updatedState.code = code;
    }
    if (nextProps.scope !== prevState.scope) {
      const scope = getEnhancedScope(nextProps.scope);
      updatedState.scope = scope;
    }
    return isEmpty(updatedState) ? null : updatedState;
  }
```

### Possible Fix

A possible fix could be to enhance the scope on render when passed to the `react-live` components. This would make it possible to compare the scope stored in the state with the new one. (Although I really do not see any good reason for the `ComponentPlayground` to store the code and scope in state anyway.)

I you like I could create a PR with a possible fix. Already created a working ComponentPlayground component for my upcoming presentation.

**Comment by lennerd** [comment]
I think the ComponentPlayground should actually be an uncontrolled component, right? Mixing both concepts in my experience is a very hard task leading to a lot of pitfalls and very complex components.

In the case of ComponentPlayground its currently not even working correctly when using the examples in this repo. Simply because the component overwrites changes to the code (in state) with the default code snippet.

**Comment by mighty-titan** [comment]
Any update at this? This is still a problem and @lennerd  solution works for the moment.

**Comment by dmitrif** [comment]
Any update on this? As of right now `ComponentPlayground` can be considered broken.

**Comment by angelo-v** [comment]
Really sad, that such an outstanding feature is not usable at the moment

**Comment by gkemp94** [comment]
Does anybody know when this was broken so I can roll back?

**Comment by kale-stew** [comment]
This has been fixed by #800.
